### ISSUE TITLE
Incorrect I18n.locale inside route

### ISSUE THREAD

**Issue by elct9620** [litaio/lita#225]
Hi, I have some problems when the message is `route` to the specific method.

My handler:

```ruby
class Deployment < Handler
  route(/^deploy (?<branch>.+)/, :deploy, command: true, help: {
    t("help.key") => t("help.value"), # Got `:'zh-TW`
  })

  def deploy(response)
    response.reply(I18n.locale) # => Got `:en`
  end
end
```

The correct one is `:zh-TW` because the environment variable `LANG="zh-TW"` but it reset to `:en` when it inside the response method.

I have no idea why it changed, I add some monkey patch to detect `I18n.locale=` but it didn't changed after Lita is booted.

**Comment by jimmycuadra** [comment]
Can you share the code for your monkey patch? New threads are created to dispatch messages to handlers, so it's possible the i18n library uses some thread local data that isn't being accounted for.

**Comment by elct9620** [comment]
I create a `bin/lita` as an executable script to verify the `I18n.locale=` is be called.

```ruby
#!/usr/bin/env ruby

require 'i18n'

I18n.module_eval do
  class << self
    alias_method :origin_locale=, :locale=

    def locale=(new_locale)
      puts "Locale is set to #{new_locale}"
      pp caller
      self.origin_locale = new_locale
    end
  end
end

Bundler.require

Lita.run(Bunder.root.join("lita_config.rb"))
```

But I can see the `Locale is set to zh-TW` twice when the config ls loaded and `ENV['LANG']` is detected.

Therefore I have no idea why the `I18n.locale` is different when the method is called. 🤔

**Comment by jimmycuadra** [comment]
From reading https://guides.rubyonrails.org/i18n.html it seems that `I18n.locale` is local to the current thread, but `I18n.default_locale` is not. Can you try adding `I18n.default_locale = "zh-TW"` at the very end of your lita_config.rb file (after the `Lita.configure` call)? If that works, I think it will be necessary to add a new `default_locale` method to the configuration, as the regular "locale" doesn't work how I expected, and changing it would technically be a breaking change.

**Comment by elct9620** [comment]
After append `I18n.default_locale` to the `lita_config.rb` the locale is correct set the `:zh-TW` in the chatbot response.

**Comment by jimmycuadra** [comment]
I just released [v4.8.0](https://github.com/litaio/lita/releases/tag/v4.8.0), which fixes this: Setting the `LANG` environment variable will now set `I18n.default_locale` in addition to `I18n.locale`. Alternatively, you can set the default locale manually in your config file with the new `config.robot.default_locale` configuration attribute.
